test/connection.test.js

```diff
@@ -59,6 +59,38 @@ describe('connections', function() {
     });
   });
 
+  it('should disconnect then connect and ORM should work', function(done) {
```

Comment:
Let's rework this test based on Promises and be explicit in checking `ds.connected` value. I also used a new datasource instance to not pollute `db` used by other tests:

```js
  it('should disconnect then connect and ORM should work', function() {
    var ds = new DataSource(mysqlConnector, config);
    var Student = ds.define('Student', {
      name: {type: String, length: 255},
      age: {type: Number},
    }, {
      forceId: false,
    });

    return ds.connect()
      .then(function(err) {
        should.not.exist(err);
        return ds.automigrate(['Student']);
      })
      .then(function(err) {
        should.not.exist(err);
        should(ds.connected).be.True();
        return ds.disconnect();
      })
      .then(function(err) {
        should.not.exist(err);
        should(ds.connected).be.False();
        return ds.connect();
      })
      .then(function(err) {
        should.not.exist(err);
        should(ds.connected).be.True();
        return Student.count();
      })
      .then(function(count) {
        should(count).be.a.Number();
        return ds.disconnect();
      });
  });
```

Reply:
Looks great, thanks. I didn't use promises to be compliant to the rest of the code.

Reply:
Yea that's a good point, but mocha supports both flavours and this is more elegant way of writing it. Also, before these changes, the linter was complaining of the `async function` callback you had for `automigrate` call (you can see this by running `npm run lint`).

Reply:
I know, thanks for pointing me out. I always use async/await and promises in my code, also for testing using mocha. Sometimes, contributing to a project is a matter of adapting own personal style and conventions to those of the project. Anyway, your changes are great, thanks.

Reply:
I went into more details because I thought you were not convinced haha. I didn't know you were expecting me to make the changes. I was expecting you to apply them 😄. I'll be happy to make the changes but given that you have them in your `master` branch, I might not be able to do so. I'll give it a try and report back.